Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -786,7 +786,13 @@ void GraphicsContextSkia::setLineDash(const DashArray& dashArray, float dashOffs
         return;
     }
 
-    m_skiaState.m_stroke.dash = SkDashPathEffect::Make(dashArray.data(), dashArray.size(), dashOffset);
+    if (dashArray.size() % 2 == 1) {
+        // Repeat the array to ensure even number of dash array elements, see e.g. 'stroke-dasharray' spec.
+        DashArray repeatedDashArray(dashArray);
+        repeatedDashArray.appendVector(dashArray);
+        m_skiaState.m_stroke.dash = SkDashPathEffect::Make(repeatedDashArray.data(), repeatedDashArray.size(), dashOffset);
+    } else
+        m_skiaState.m_stroke.dash = SkDashPathEffect::Make(dashArray.data(), dashArray.size(), dashOffset);
```

Comment:
Looks good, I've noticed that in cairo (and chromium does the same) we also check that the array is not all 0 (for skia we only check the array is not empty) and set the dash to nullptr in that case too. We can do that in a follow up patch.

Reply:
Good point - skia doesn't validate all-0 case.